service/client_state.cc

```diff
@@ -254,6 +257,20 @@ future<> service::client_state::ensure_exists(const auth::resource& r) const {
     });
 }
 
+void service::client_state::ensure_alterable_permisssions(const auth::resource& r) const {
+    // We assume we have already checked that the resource exists.
+    if (r.kind() == auth::resource_kind::functions) {
+        auth::functions_resource_view v(r);
+        const auto keyspace = v.keyspace();
+        const auto function_signature = v.function_signature();
```

Comment:
silly nitpick: I think it doesn't help to add these shortcuts variables, each of them only used in one place.